Feat/hero content image and blog hero image

ikrestov opened the pull request:
cc @moaiandin

tbarn commented:
Should we include a slight darkening of the image so we aren't limited by what images we use and so the subtitle is still easily readable? 

<img width="1440" alt="Screen Shot 2019-04-08 at 11 48 11 AM" src="https://user-images.githubusercontent.com/2008034/55742049-59808d00-59f4-11e9-9434-ffa4408af2af.png">

moaiandin commented:
Hi @tbarn, we are already darkening them, could make it more dark but then it will look a little bit too dark.

lottamus commented:
![image](https://user-images.githubusercontent.com/7423098/55759861-1d632180-5a20-11e9-8f64-fc74f11ddf95.png)
Looks like it failed to build. Can you try running a `yarn build` locally?

lottamus commented on src/components/Hero/index.tsx:
```diff
@@ -364,6 +249,145 @@ export const Hero: React.FunctionComponent<IHero> = ({
+                backgroundSize: 'cover',
+              }}
+            >
+              {contentBgOverlay && (<div
```
I don't think we need this extra div

lottamus commented on src/components/Hero/index.tsx:
```diff
@@ -364,6 +249,145 @@ export const Hero: React.FunctionComponent<IHero> = ({
+
+        <div className={cn('relative', 'w-100')}>
+          
+          {contentBgImage && (
```
Let's move the contentBgImage outside of the div on L255: https://github.com/stoplightio/stoplight.io/pull/737/files#diff-46589db87698917798337df55156d320R255 so the image fills the header as well

lottamus commented on src/components/Hero/index.tsx:
```diff
@@ -364,6 +249,145 @@ export const Hero: React.FunctionComponent<IHero> = ({
+          {contentBgImage && (
+            <div
+              className={cn('absolute pin z-0')}
+              style={{
```
Change this to be something like:

```
backgroundImage: `linear-gradient(rgba(0, 0, 0, 0.5), rgba(0, 0, 0, 0.5)), url("${contentBgImage}")`,
backgroundSize: 'cover',
backgroundRepeat: 'no-repeat',
backgroundPosition: '50% 50%',
```

lottamus commented on src/components/Hero/index.tsx:
```diff
@@ -364,6 +249,145 @@ export const Hero: React.FunctionComponent<IHero> = ({
+
+        <div className={cn(headerHeightClass, 'w-100')} />
+
+        <div className={cn('relative', 'w-100')}>
```
Lets add a text shadow something like:

```
textShadow: rgba(0, 0, 0, 0.5) 1px 1px 0px
```

ikrestov commented:
Hey,

I have fixed typo breaking build, rebased and addressed comments.

![image](https://user-images.githubusercontent.com/1066322/55791103-b1cb9380-5ab5-11e9-86d4-cab530b0a235.png)

Thanks,
Igor.

moaiandin commented:
@lottamus all fixed and ready to go :)
